core/src/main/resources/spline.default.properties

```diff
@@ -52,6 +52,7 @@ spline.lineageDispatcher.http.className=za.co.absa.spline.harvester.dispatcher.H
 spline.lineageDispatcher.http.producer.url=
 spline.lineageDispatcher.http.timeout.connection=2000
 spline.lineageDispatcher.http.timeout.read=120000
+spline.lineageDispatcher.http.securityHeader.key=somekeys
```

Comment:
the security headers should not be mandatory, but optional (as apiVersion and requestCompression is)

also is probably better to not provide any default value here

Reply:
I wanted the same but i am not sure how we can pass the n number of header as request compression and apiversion is added as argument to HTTPDispather class.

Reply:
Is it ok to create a seperate HttpLineageDispather for Azure functions like for open lineage is done. I see the openLineageHttpDispatcher is dependent on RestClient from Httpdispather. This will solve my team problem and will also not be a big change

Reply:
> I wanted the same but i am not sure how we can pass the n number of header as request compression and apiversion is added as argument to HTTPDispather class.

Assuming the config is in this format:
```
spline.lineageDispatcher.http.header.customName.key=X-FUNCTIONS_KEY
spline.lineageDispatcher.http.header.customName.value=somekeys

spline.lineageDispatcher.http.header.customName2.key=SecondHeaderKey
spline.lineageDispatcher.http.header.customName2.value=SecondHeaderValue
```
You can do the extraction in `HttpLineageDispatcherConfig` class like : 
```
val headersConfig = config.subset("header")
```
then you can do `headersConfig.getKeys` and for each header get the key and value and put them in the scala map. That map can be a dispatcher parameter. `Map[String, String]`

Reply:
> Is it ok to create a seperate HttpLineageDispather for Azure functions like for open lineage is done. I see the openLineageHttpDispatcher is dependent on RestClient from Httpdispather. This will solve my team problem and will also not be a big change

It's ok to create as many dispatchers as you want, but if you want them to become part of Spline, they must be generic enough to be useful for others.

If only difference is one header it means there is lot of functionality duplicated.

If you see there is a lot of things needed for Azure that are different from the standard dispatcher it would make sense to do AzureDispatcher, but in case of adding headers I think it would be useful to have them in generic http dispatcher anyway.